packages/xo-web/src/xo-app/xoa/cloud-config/index.js

```diff
@@ -0,0 +1,132 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import addSubscriptions from 'add-subscriptions'
+import decorate from 'apply-decorators'
+import Icon from 'icon'
+import React from 'react'
+import { confirm } from 'modal'
+import { getApplianceId, subscribeCloudXoConfig, subscribeCloudXoConfigBackups } from 'xo'
+import { groupBy, sortBy } from 'lodash'
+import { injectState, provideState } from 'reaclette'
+import { SelectXoCloudConfig } from 'select-objects'
+
+import BackupXoConfigModal from './Backup-xo-config-modal'
+
+const CloudConfig = decorate([
+  addSubscriptions({
+    cloudXoConfig: subscribeCloudXoConfig,
+    cloudXoConfigBackups: subscribeCloudXoConfigBackups,
+  }),
+  provideState({
+    initialState: () => ({ config: undefined }),
+    effects: {
+      downloadCloudXoConfig:
+        () =>
+        ({ config, isConfigDefined }) => {
+          if (isConfigDefined) {
+            window.open(config.content, '_blank')
+          }
+        },
+      importCloudXoConfig:
+        () =>
+        async ({ config, isConfigDefined }) => {
+          if (isConfigDefined) {
+            const resp = await fetch(`./rest/v0/cloud/xo-config/backups/${config.id}/actions/import?sync`, {
+              method: 'POST',
+            })
+            if (!resp.ok) {
+              throw new Error(resp.statusText)
+            }
+            return {
+              config: undefined,
+            }
+          }
+        },
+      onChangeCloudXoConfig: (_, config) => ({
+        config,
+      }),
+      toggleEnableCloudXoConfig:
+        () =>
+        async (state, { cloudXoConfig }) => {
+          let passphrase
+          if (!cloudXoConfig.enabled) {
+            const params = await confirm({
+              icon: 'backup',
+              title: _('cloudXoConfig'),
+              body: <BackupXoConfigModal />,
+            })
+            passphrase = params.passphrase
+          }
+
+          const resp = await fetch('./rest/v0/cloud/xo-config', {
+            method: 'PATCH',
+            headers: {
+              'Content-Type': 'application/json',
+            },
+            body: JSON.stringify({
+              enabled: !cloudXoConfig.enabled,
+              passphrase,
+            }),
+          })
+          if (!resp.ok) {
+            throw new Error(resp.statusText)
+          }
+          subscribeCloudXoConfig.forceRefresh()
+        },
+    },
+    computed: {
+      applianceId: async () => {
+        const { id } = await getApplianceId()
+        return id
+      },
+      groupedConfigs: ({ applianceId, sortedConfigs }) =>
+        sortBy(groupBy(sortedConfigs, 'xoa_id'), config => (config[0].xoa_id === applianceId ? -1 : 1)),
```

Comment:
Should we add a second sorting criteria to ensure that it's always sorted consistently?

Reply:
It could be literally anything as long as it's consistent. Even sorting on the ID. But if the collection is already sorted when we receive it, I guess it's ok not to sort it again?

Reply:
The collection we receive is already sorted.